### PR TITLE
Make binary recording memmap efficient III (Avoiding spikes by only reserving memory equal to the trace)

### PR DESCRIPTION
**Pull request by h-mayorquin**
This is a follow up from #1742, it only allocates opens a memmap pointing to the specific trace and not to the whole file.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Pytest runs all of their tests on the same process and the function that I have measures peak memory within the process so of course this failing. Thinking on how to best handles this.

**Comment by h-mayorquin** [comment]
OK, with the recent merge of https://github.com/SpikeInterface/spikeinterface/pull/2796 I got the willpower to finish this off. Test should be passing now and I fixed the merge conflicts. 

@alejoe91 @samuelgarcia same drills as in #2796 but this time for reading.

File: src/spikeinterface/core/binaryrecordingextractor.py [hunk 1]
```diff
@@ -201,23 +193,51 @@ def get_traces(
...
-        length = self.memmap_length
-        memmap_offset = self.memmap_offset
+        if start_frame is None:
+            start_frame = 0
+
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+
+        if end_frame > self.get_num_samples():
+            raise ValueError(f"end_frame {end_frame} is larger than the number of samples {self.get_num_samples()}")
```
Comment (samuelgarcia) [inline src/spikeinterface/core/binaryrecordingextractor.py:203]: This now handled outside
```suggestion
```

**Comment by samuelgarcia** [comment]
@h-mayorquin I guess this is ready for merging.
This is OK for me I made a proposal for the boundaries.

**Comment by h-mayorquin** [comment]
@samuelgarcia yes, makes sense. It was an old PR : )

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**